src/OSPSuite.Core/Domain/Services/EntityTracker.cs

```diff
@@ -10,24 +10,26 @@ public interface IEntityTracker
 
    public class EntityTracker : IEntityTracker
    {
-      private readonly IObjectPathFactory _objectPathFactory;
-      private readonly IObjectTypeResolver _objectTypeResolver;
+      private readonly IEntityPathResolver _entityPathResolver;
 
-      public EntityTracker(IObjectPathFactory objectPathFactory, IObjectTypeResolver objectTypeResolver)
+      public EntityTracker(IEntityPathResolver entityPathResolver)
       {
-         _objectPathFactory = objectPathFactory;
-         _objectTypeResolver = objectTypeResolver;
+         _entityPathResolver = entityPathResolver;
       }
 
       public void Track(IEntity entityToTrack, IEntity sourceBuilder, SimulationBuilder simulationBuilder)
       {
-         var sourcePath = _objectPathFactory.CreateAbsoluteObjectPath(sourceBuilder).ToString();
-         var sourceType = _objectTypeResolver.TypeFor(sourceBuilder);
+         var sourcePath = sourcePathFor(sourceBuilder);
+
+         if (entityToTrack.Name == "Ontogeny factor")
+         {
+            entityToTrack.Name = entityToTrack.Name;
+         }
```

Comment:
you don't need the curly braces

Reply:
lol this needs to be removed anyways. It was for debug purpose so that I could set a breakpoint

Reply:
lol, oh yeah...